### ISSUE TITLE
Websocket not working at non-root path

### ISSUE THREAD

**Issue by adpavlov** [ansible/awx#12241]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Summary

Changes from #11342 and #652 are not full

### AWX version

21.0.0

### Select the relevant components

- [X] UI
- [ ] API
- [ ] Docs

### Installation method

kubernetes

### Modifications

no

### Steps to reproduce

Deploy AWX with custom `ingress_path: /awx`

### Expected results

websocket should work

### Actual results

`2022-05-17 08:46:41,031 ERROR    [-] daphne.ws_protocol [Failure instance: Traceback: <class 'ValueError'>: No route found for path 'awx/websocket/'.
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/autobahn/websocket/protocol.py:2841:processHandshake
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/txaio/tx.py:366:as_future
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/twisted/internet/defer.py:151:maybeDeferred
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/daphne/ws_protocol.py:72:onConnect
--- <exception caught here> ---
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/twisted/internet/defer.py:151:maybeDeferred
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/daphne/server.py:201:create_application
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/routing.py:54:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/sessions.py:47:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/sessions.py:145:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/sessions.py:169:__init__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/middleware.py:31:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/routing.py:150:__call__
]
2022-05-17 08:46:41,031 ERROR    [Failure instance: Traceback: <class 'ValueError'>: No route found for path 'awx/websocket/'.
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/autobahn/websocket/protocol.py:2841:processHandshake
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/txaio/tx.py:366:as_future
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/twisted/internet/defer.py:151:maybeDeferred
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/daphne/ws_protocol.py:72:onConnect
--- <exception caught here> ---
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/twisted/internet/defer.py:151:maybeDeferred
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/daphne/server.py:201:create_application
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/routing.py:54:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/sessions.py:47:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/sessions.py:145:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/sessions.py:169:__init__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/middleware.py:31:__call__
/var/lib/awx/venv/awx/lib64/python3.9/site-packages/channels/routing.py:150:__call__`

### Additional information

It seems that issue is in  https://github.com/ansible/awx/blob/48b016802c517ff04d1cff4c43e64f17bb77a7a8/awx/main/routing.py

```
websocket_urlpatterns = [
    re_path(r'websocket/$', consumers.EventConsumer),
    re_path(r'websocket/broadcast/$', consumers.BroadcastConsumer),
]
```
From https://docs.djangoproject.com/en/4.0/ref/urls/:

When a route ends with $ the whole requested URL, matching against path_info, must match the regular expression pattern (re.fullmatch() is used).

Replacing with

```
websocket_urlpatterns = [
    re_path(r'websocket/', consumers.EventConsumer),
    re_path(r'websocket/broadcast/', consumers.BroadcastConsumer),
]
```

solves the issue

**Comment by nixocio** [comment]
~May be related to #12097~

**Comment by adpavlov** [comment]
> May be related to #12097 

No, it's not. I believe I found the root cause, please check pull request

**Comment by DrBautze** [comment]
Issue persists with 21.7.0. Any update on this?

**Comment by ugurserhattoy** [comment]
> Issue persists with 21.7.0. Any update on this?

Did you find any fix? 
I tried adding these annotations to deployment yaml. Didn't work.
![image](https://user-images.githubusercontent.com/54373170/202015853-0a5b330f-7f49-44f5-9ab7-1cf76e5b7812.png)

**Comment by adpavlov** [comment]
> > Issue persists with 21.7.0. Any update on this?
> 
> 
> 
> Did you find any fix? 
> 
> I tried adding these annotations to deployment yaml. Didn't work.
> 
> ![image](https://user-images.githubusercontent.com/54373170/202015853-0a5b330f-7f49-44f5-9ab7-1cf76e5b7812.png)
> 
> 

Check related commit. For me it solved the issue

**Comment by shanemcd** [comment]
Sorry this fell through the cracks for so long! I rebased and merged your PR. It will go out in the next release of AWX.

**Comment by shanemcd** [comment]
Reopening since this got reverted...

**Comment by adpavlov** [comment]
Added https://github.com/ansible/awx/pull/13645#issuecomment-1458232725

**Comment by mabashian** [comment]
@TheRealHaoLiu are we planning on taking another crack at this or can you provide some context around the revert and possible paths forward?

**Comment by adpavlov** [comment]
bump the issue. Since AWX 22.0.0 my fix won't help because path is even hardcoded and and not taking into consideration non-root path @TheRealHaoLiu @shanemcd 

https://github.com/ansible/awx/blob/2db29e5ce214699cee8551f21da3c687a9f4fe79/awx/main/wsrelay.py#L73
